### PR TITLE
feat: init function refactor

### PR DESCRIPTION
**Pull request by hexqi**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Background and solution

- 优化init方法，删除重复逻辑
- demo设计器改用vite命令，改用8090端口
- init方法支持传入应用生命周期函数

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Introduced Vite for serving and building the application, enhancing performance and development experience.
  - Added Vite configuration file to customize server settings.

- **Improvements**
  - Updated copyright notices.
  - Simplified initialization process by consolidating steps into a single `init()` function.
  - Enhanced `dev:demo` script to include setup steps before running services.

- **Refactor**
  - Moved `init` function to a separate file for better modularity and maintainability.
  - Updated import paths and function calls for improved code structure.

- **Chores**
  - Added `"type": "module"` to `package.json` for `vite-plugin-generate-comments`.
  - Updated package dependencies in setup script.

### CODE REVIEW

File: packages/design-core/src/init.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import TinyThemeTool from '@opentiny/vue-theme/theme-tool'
+import { tinySmbTheme } from '@opentiny/vue-theme/theme' // SMB 主题
+import { defineEntry } from '@opentiny/tiny-engine-entry'
+import App from './App.vue'
+import defaultRegistry from '../registry.js'
+
+import 'virtual:svg-icons-register'
+
+const type = (obj) => {
+  return Object.prototype.toString.call(obj).match(/\[object (.*)\]/)[1]
```
Comment (chilingling) [inline packages/design-core/src/init.js]: 这个放到 utils ？

File: designer-demo/vite.config.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { defineConfig } from 'vite'
+import getDefaultConfig from '@opentiny/tiny-engine/vite.config.js'
+
+export default defineConfig((options) => {
+  const defaultConfig = getDefaultConfig(options)
```
Comment (chilingling) [inline designer-demo/vite.config.js:5]: 这块感觉可以做个函数，用户传入 options 进来，我们主动去 merge 用户的 options，而不是用户来处理 merge option 的动作。
比如：

```javascript
export default defineConfig((options) => {
    const userOptions = { ... }
    return mergeWithEngineOption(options, userOptions);
})
```

File: packages/design-core/src/init.js (hunk 1 above)
Comment (hexqi) [reply packages/design-core/src/init.js]: 已修改

File: designer-demo/vite.config.js (hunk 2 above)
Comment (hexqi) [reply designer-demo/vite.config.js:5]: 已修改。vite有提供merge函数

File: packages/design-core/src/init.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+    // eslint-disable-next-line no-new
+    new TinyThemeTool(tinySmbTheme, 'smbtheme') // 初始化主题
+
+    if (import.meta.env.VITE_ERROR_MONITOR === 'true' && import.meta.env.VITE_ERROR_MONITOR_URL) {
+      initMonitor(import.meta.env.VITE_ERROR_MONITOR_URL)
+    }
+
+    window.TinyGlobalConfig = globalConfig
+    setGlobalConfig(globalConfig)
```
[the remark this answers is not in the document]
Comment (hexqi) [reply packages/design-core/src/init.js:74]: 遗留Todo

**Review by chilingling — APPROVED** [review]

**Review by kevinmoch — APPROVED** [review]

**Pull request merged**